src/modules/formBuilder/components/itemEditor/pickLists/PickListOption.tsx

```diff
@@ -0,0 +1,111 @@
+import { Stack } from '@mui/material';
+import { useCallback } from 'react';
+import ControlledCheckbox from '@/modules/form/components/rhf/ControlledCheckbox';
+import ControlledTextInput from '@/modules/form/components/rhf/ControlledTextInput';
+import { FormItemControl } from '@/modules/formBuilder/components/itemEditor/types';
+import { Component } from '@/types/gqlTypes';
+
+interface PickListOptionProps {
+  control: FormItemControl;
+  index: number;
+  formItemComponent: Component;
+  setPickListInitialSelected: (code: string) => void;
+  isCodeUnique: (coe: string) => boolean;
+  code?: string;
+}
+
+const PickListOption: React.FC<PickListOptionProps> = ({
+  control,
+  code,
+  index,
+  formItemComponent,
+  isCodeUnique,
+  setPickListInitialSelected,
+}) => {
+  const onChangeInitialSelected = useCallback(
+    (value: boolean) => {
+      // Even though it's a controlled RHF checkbox, we still provide an onChange callback
+      // in order to set all the other pick list options' `initialSelected` values to false when this one is set to true.
+      if (value && code) setPickListInitialSelected(code);
+    },
+    [setPickListInitialSelected, code]
+  );
+
+  return (
+    <Stack gap={2}>
+      <ControlledTextInput
+        control={control}
+        name={`pickListOptions.${index}.code`}
+        label='Code'
+        helperText='Must be unique'
+        required={true}
+        rules={{
+          validate: (input) => isCodeUnique(input),
+        }}
+      />
+      <ControlledTextInput
+        control={control}
```

Comment:
Originally I had wanted to hide a lot of these options under an "advanced" slideout/collapse panel. It's a bit complicated though, because using `replace` (which we do in the onChangeInitialSelected callback) causes the whole thing to re-render, so we would need to either remove that logic, or keep track in a separate piece of state of which advanced panels were open.

I'm ok either way for v1, let me know your preference.

Reply:
yeah fair enough!! sorry the "initial selected" caused some complexity and it's not even working :\ I can't say confidently whether it's used anywhere or not.

Reply:
Ok update, after [these changes](https://github.com/greenriver/hmis-frontend/pull/850#discussion_r1681533408) implementing the collapsible advanced panel turned out to be easy. Let me know if you think it looks okay -- maybe as currently implemented (with an icon button) it's too unobtrusive? I've also tried implementing it as a SimpleAccordion, (commit on separate branch https://github.com/greenriver/hmis-frontend/commit/eca9e231ee36a751fd348313fa4689ec414a45a9). It's nice to reuse another pattern we have elsewhere, but it's pretty bulky, which loses part of the ux benefit.